hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -755,21 +763,30 @@ onOpenNetworkReqDec env ledger ttl openState decommitTx =
   waitOnApplicableDecommit cont =
     case mExistingDecommitTx of
       Nothing ->
-        case applyTransactions currentSlot localUTxO [decommitTx] of
-          Right utxo' -> cont utxo'
-          Left (_, validationError)
-            | ttl > 0 ->
-                wait $
-                  WaitOnNotApplicableDecommitTx
-                    ServerOutput.DecommitTxInvalid{localUTxO, validationError}
-            | otherwise ->
-                cause . ClientEffect $
-                  ServerOutput.DecommitInvalid
-                    { headId
-                    , decommitTx
-                    , decommitInvalidReason =
-                        ServerOutput.DecommitTxInvalid{localUTxO, validationError}
-                    }
+        if utxoFromTx decommitTx == mempty
+          then
+            cause . ClientEffect $
+              ServerOutput.DecommitInvalid
+                { headId
+                , decommitTx
+                , decommitInvalidReason =
+                    ServerOutput.DecommitTxInvalid{localUTxO, validationError = emptyDecommitError}
```

Comment:
It feels ugly that you "shoehorn" the hard-coded `emptyDecommitError` into a ledger `validationError` field here.

But this makes me think.. is this whole thing even an issue if we use a real ledger? i.e. wouldn't the `cardanoLedger` make empty decommits impossible by requiring transactions to be balanced?

Must: Validate whether the #1502 issue is even real by having a test that uses real `Tx` trying to do decommit nothing. What's the behavior without these changes?

Reply:
The ledger prevents empty tx outputs with `ValueNotConserved` and in case you don't provide any inputs then it is `TxBodyEmptyTxIns`. So this is actually not an issue we thought it would be :+1: